tests/test_weaviate_adaptor_live.py

```diff
@@ -0,0 +1,77 @@
+"""Optional live Weaviate tests mirroring the benchmark flow."""
+
+from __future__ import annotations
+
+import numpy as np
+import pytest
+from datasets import Dataset
+
+from inatinqperf.adaptors.base import Query
+from inatinqperf.adaptors.metric import Metric
+from inatinqperf.benchmark.benchmark import Benchmarker
+from inatinqperf.benchmark.configuration import (
+    Config,
+    DatasetConfig,
+    EmbeddingParams,
+    SearchParams,
+    VectorDatabaseConfig,
+    VectorDatabaseParams,
+)
+
+BASE_URL = "http://localhost:8080"
+
+
+@pytest.mark.skipif(True, reason="Live Weaviate test disabled by default")
+def test_weaviate_live_single_vector(tmp_path) -> None:
+    dataset = Dataset.from_dict(
+        {
+            "id": [123],
+            "embedding": [np.random.default_rng(0).standard_normal(512).astype(np.float32).tolist()],
+        }
+    )
+    # print("Dataset row:", dataset[0])
+
+    config = Config(
+        dataset=DatasetConfig(
+            dataset_id="stub",
+            splits="validation",
+            directory=tmp_path / "raw",
+            export_images=False,
+        ),
+        embedding=EmbeddingParams(
+            model_id="stub",
+            batch_size=1,
+            directory=tmp_path / "emb",
+        ),
+        vectordb=VectorDatabaseConfig(
+            type="weaviate.hnsw",
+            params=VectorDatabaseParams(
+                metric=Metric.COSINE,
+                url=BASE_URL,
+                class_name="BenchmarkLiveTest",
+                timeout=10.0,
+            ),
+        ),
+        search=SearchParams(
+            topk=1,
+            queries_file=tmp_path / "queries.txt",
+        ),
+        update={"add_count": 1, "delete_count": 1},
+    )
+
+    benchmarker = Benchmarker.__new__(Benchmarker)
```

Comment:
Huh?? Why not just `benchmarker = Benchmarker()`? Why this very strange way?

Reply:
Please don't mark comments as resolved unless there is some actual change addressing the review comment or at least a valid reason provided.